examples/twittermap/web/public/javascripts/mapresultcache/services.js

```diff
@@ -25,7 +26,22 @@ angular.module('cloudberry.mapresultcache', ['cloudberry.common'])
             city: 'I'
         });
         const INVALID_VALUE = 0;
+    
 
+        /** 
+        * preloadRequest if this.isNewKeyword only.
+        * current cache is cleared when new keyword is requested.
+        */
+        this.isNewKeyword = function (keywords) {
```

Comment:
The cache needs to be cleared if: (1), or (2) time interval is changed for _non-county interval_.
countyStore cache county data in the entire time range. When the time interval is changed, method getValues[line 100] and getCountyTimeSeriesValues[line131] find the new interval from countyStore, thus reducing the number of requests.
The clear cache decisions are made in getGeoIdsNotInCache. The cache cleared above is to avoid displaying "old" cached data, in the case ws.onmessage() is not received in the order that the user sends them.